### ISSUE TITLE
Cloud Variables Don't Save on Server

### ISSUE THREAD

**Issue by memer3638** [SheepTester/primitive-cloud-server#21]
Hello, i'm using this server for testing and when i used this and started it up it says it was on but didn't save anything onto the server which was weird and i'm wondering what's causing it. i'm using it and turning a project into the turbowarp packager.

**Comment by SheepTester** [comment]
it's possible it's failing to connect? could check the inspect element console for details. Scratch will set the cloud variable value locally even if it fails to connect or update on the server

**Comment by memer3638** [comment]
> No, it is connected and online but doesn't store data onto it

**Comment by memer3638** [comment]
> it's possible it's failing to connect? could check the inspect element console for details. Scratch will set the cloud variable value locally even if it fails to connect or update on the server

No, it is online and up

**Comment by SheepTester** [comment]
being online doesn't mean the project can connect to it

**Comment by memer3638** [comment]
> being online doesn't mean the project can connect to it

And that's about my problem it won't connect or it can't save variables onto it

**Comment by SheepTester** [comment]
is there an error in the inspect element console

**Comment by KianiDev** [comment]
Try with a WebSocket online tester and send a request for it. 
{ "method": "handshake", "project_id": "Test1234" }
{ "method": "set", "name": "☁ example", "value": "1234" }
A JSON file should be created in "cloud-vars" folder, which has the data of "☁ example" variable.
If this doesn't happen, then it might be a server problem.

**Comment by memer3638** [comment]
> is there an error in the inspect element console

Yeah but I used other cloud servers like turbowarps and they show the same errors but still work completely fine

**Comment by zaid4412100** [comment]
I think the issue was that you where using "wss://" instead of "ws://". "wss://" is for secure web-socket connections (doesn't work because there are no keys or certificates). I also had the issue, but I copy-pasted it when I tried to use the TurboWarp's parameters (?cloud_host=). I went and looked how it worked, then I realized I used a secure connection.

**Comment by TriangularDev** [comment]
Where is the server hosted on? Some hosting providers don't work with the server.

**Comment by zaid4412100** [comment]
> Where is the server hosted on? Some hosting providers don't work with the server.

He is using testing it

**Comment by KianiDev** [comment]
1. Try both WS and WSS protocols. WSS is gonna connect when you run it on a VM and WS is when you run it on a computer yourself. There are no keys and certificates when hosting it yourself.
2. If you're running it on a VM, try to test it in the way I mentioned earlier.
3. If you're running it on your computer, use the localhost address first. Sometimes the other URLs provided need extra steps to work properly.

Also, does anything happen when you try to connect to it? Any errors, files created, or anything in the logs? If yes, please send them here.

**Comment by KianiDev** [comment]
If you are hosting it on your computer, make sure that TurboWarp and the server both have access in the firewall settings to local networks. It could be the Windows firewall blocking internal connections.
